### ISSUE TITLE
Panic when adding cheats

### ISSUE THREAD

**Issue by rtoIedo** [denisidoro/navi#292]
**Describe the bug**
An error occurs when adding cheats for the first time

**To Reproduce**
1. execute: `brew install denisidoro/tools/navi`
2. execute: `navi repo add denistore/cheat`
3. Error
```shell
$ brew install denisidoro/tools/navi
$ navi repo add denistore/cheat
Cloning https://github.com/denistore/cheat into /Users/rtoledo/Library/Application Support/navi/cheats/tmp...

thread 'main' panicked at 'failed to clone: unsupported URL protocol; class=Net (12)', src/flows/repo.rs:57:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Expected behavior**
Cheats should be added

**Versions:**
 - OS: macOS 10.15.1 (19B2093)
 - GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19) Copyright (C) 2007 Free Software Foundation, Inc.

**Comment by denisidoro** [comment]
Strange... It's working here, for example: https://www.katacoda.com/denisidoro/scenarios/navi

You have git installed, don't you?

**Comment by denisidoro** [comment]
Ah, there's a typo: `navi repo add denistore/cheat` should be `navi repo add denisidoro/cheats`

**Comment by rtoIedo** [comment]
Sorry, it was `navi repo add denisidoro/cheats`. I missed the `s` when I copied and pasted the command.
I'm currently in the latest version: 2.2.0. Actually, I ran these commands just a few minutes ago.

**Comment by denisidoro** [comment]
Could you share the error message for the correct command, please?

**Comment by rtoIedo** [comment]
The error message is the one I posted originally:

```
$ navi repo add denistore/cheat
Cloning https://github.com/denistore/cheat into /Users/rtoledo/Library/Application Support/navi/cheats/tmp...

thread 'main' panicked at 'failed to clone: unsupported URL protocol; class=Net (12)', src/flows/repo.rs:57:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Apparently, it has something to do with my `.gitconfig` file

```
$ mv .gitconfig .gitconfig.bak
$ navi repo add denistore/cheats
Cloning https://github.com/denistore/cheats into /Users/rtoledo/Library/Application Support/navi/cheats/tmp...

thread 'main' panicked at 'failed to clone: remote authentication required but no callback set; class=Net (12)', src/flows/repo.rs:57:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

$ mv .gitconfig.bak .gitconfig
$navi repo add denistore/cheats
Cloning https://github.com/denistore/cheats into /Users/rtoledo/Library/Application Support/navi/cheats/tmp...

thread 'main' panicked at 'failed to clone: unsupported URL protocol; class=Net (12)', src/flows/repo.rs:57:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by denisidoro** [comment]
But in the error message I still see `denistore`. My username is `denisidoro`. Am I missing something?

**Comment by rtoIedo** [comment]
You're right. Part of the problem was that typo (I also tried installing cheats from within the navi) However, the other part was my `.gitconfig` file.
I had this config in place. Just as a reference for future issues

```
[user]
	name = Rodolfo Toledo
	email = rtoledo@mail.com
[credential]
	helper = cache
[url "ssh://git@github.com/"]
	insteadOf = https://github.com/
[core]
	autocrlf = input
```

You can close the ticket. It seems nothing navi can do to solve this issue.

**Comment by denisidoro** [comment]
I'm planning on changing the way a repo is downloaded: instead of using git, using curl+tar. Maybe that will fix the issue. I'll leave this issue open until then.

**Comment by rtoIedo** [comment]
(very cool command BTW, now that I was able to install cheats, I tried it and it rocks!)

**Comment by athityakumar** [comment]
@denisidoro - I ran into a similar "panic" situation as I exited navi with Ctrl+C once. Any idea how I can fix this panic issue and start using navi again? 😅

**Comment by denisidoro** [comment]
@athityakumar is it the same as #300?

**Comment by sipah00** [comment]
@athityakumar I also got the  same error, were you able to fix the problem?
![image](https://user-images.githubusercontent.com/23627932/77255696-680c9680-6c8f-11ea-8078-9511d948e3bc.png)

**Comment by athityakumar** [comment]
@sipah00 - I didn't face the Resource unavailable issue, but I had a different parsing issue within the cheat files. Most probably, the rust code couldn't parse one of the cheat files in one of the cheatfile repositories - so I just removed that particular repo and got it working now. 😄 

@denisidoro - Np, my issue is resolved now 😄

**Comment by athityakumar** [comment]
@denisidoro - Btw, totally forgot to thank you. Thanks a lot for such an awesome cli utility! Being able to produce "dynamic" queries by taking input from user is just brilliant - I was using something called keep until now for this, but that was also static. Hence, converting to navi now! ❤️

**Comment by denisidoro** [comment]
@sipah00 does `git clone https://github.com/denisidoro/navi` work at your terminal? Maybe it's some firewall or something?

**Comment by denisidoro** [comment]
@athityakumar could you please share the offending cheatsheet file? Does it contain a backtick?

**Comment by sipah00** [comment]
@denisidoro `git clone https://github.com/denisidoro/navi` is working on my system, I am behind proxy. 
git clone https://github.com/denisidoro/navi
![image](https://user-images.githubusercontent.com/23627932/77286960-79d75380-6cfa-11ea-9636-b85f2fca10b0.png)

**Comment by athityakumar** [comment]
Got this error again: 😢 

```
thread 'main' panicked at 'byte index 34 is not a char boundary; it is inside '’' (bytes 33..36) of `Print the last lines of a service’s logs`', src/libcore/str/mod.rs:2154:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Failed to read /dev/tty
```

**Comment by athityakumar** [comment]
Here's the full stack trace:

```
❯❯❯ RUST_BACKTRACE=1 navi --print                                                    ─╯
thread 'main' panicked at 'byte index 34 is not a char boundary; it is inside '’' (bytes 33..36) of `Print the last lines of a service’s logs`', src/libcore/str/mod.rs:2154:5
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::str::slice_error_fail
   9: core::str::traits::<impl core::slice::SliceIndex<str> for core::ops::range::RangeTo<usize>>::index::{{closure}}
  10: navi::display::limit_str
  11: navi::display::format_line
  12: navi::parser::write_cmd
  13: navi::parser::read_all
  14: navi::fzf::call
  15: navi::flows::core::main
  16: navi::handler::handle_config
  17: navi::main
  18: std::rt::lang_start::{{closure}}
  19: std::panicking::try::do_call
  20: __rust_maybe_catch_panic
  21: std::rt::lang_start_internal
  22: main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Failed to read /dev/tty
```

**Comment by denisidoro** [comment]
@sipah00 were you able to fix your issue?

**Comment by athityakumar** [comment]
@denisidoro - Thanks for the fix - works without any hassle in v2.4.1! 🎉

**Comment by denisidoro** [comment]
Probably fixed by https://github.com/denisidoro/navi/pull/336

If this error persists after I release the 2.5+ version, please feel free to reopen the issue.
